power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/y_bus.hpp

```diff
@@ -350,6 +354,79 @@ template <bool sym> class YBus {
         admittance_ = std::make_shared<ComplexTensorVector<sym> const>(std::move(admittance));
     }
 
+    template <UpdateIsProgressive T>
+    void update_admittance_increment(std::shared_ptr<T const> const& math_model_param_incrmt,
+                                     bool is_decrement = false) {
+        const double mode = is_decrement ? -1.0 : 1.0;
+        if (!is_decrement) {
+            // overwrite the old cached parameters in increment mode (update)
+            math_model_param_incrmt_ = math_model_param_incrmt;
+        } else if (!decremented_) {
+            math_model_param_incrmt_ = {}; // reset recorded increment that is already decremented
+            return;
+        }
+
+        // construct admittance data
+        ComplexTensorVector<sym> admittance(nnz());
```

Comment:
can you please rename `nnz`? it confuses me every single time 😬

Reply:
I would say renaming it to `n_nz()` to better match popular literature but am not sure if that helps. `nnz()` was also used in a classic literature on sparse matrix.

Reply:
`nnz()` was actually existing before me, I could do an alias here to help read but I don't think I should be at the liberty renaming the original variable hidden in the source. Right?

Reply:
agreed out of scope then. this is btw the reason why i prefer descriptive variable names: you never know whether it was intentionally added or copied from another location (where it might actually make sense to use the shorter version)